CONTRIBUTING.MD

```diff
@@ -1,30 +1,62 @@
-# How to contribute
+# How to contribute 🌟
 
-1. Fork this repository.
+Thank you for your interest in contributing! We welcome all kinds of contributions, whether they are bug fixes, new features, documentation improvements, or any other suggestions. This guide will walk you through the process of contributing to FastStore framework.
 
-2. Clone your forked `faststore` repository.
+## 1. Requirements
 
-3. Create a new branch for the changes you are working on.
+Before you begin, make sure you have the following installed on your local machine to run the project:
 
-## Developing Locally
+- [Git](https://git-scm.com)
+- [Node.js](https://nodejs.org/en)
+- [Yarn](https://yarnpkg.com/)
+- [Turbo](https://turbo.build/repo/docs/getting-started/installation#installing-turbo)
+- Your Preferred Code Editor (VSCode, Sublime, etc.)
 
-### Developing Packages
+## 2. Getting Started
 
-1. Run `yarn` at the root of the repo.
+### Forking the Repository
 
-2. Run `yarn build` at the root of the repo.
+**If you don’t have write access to this repository**, you'll need to fork it (else, skip this step):
+
+Fork the repository (click the <kbd>Fork</kbd> button at the top right of
+[this page](https://github.com/vtex/faststore)).This will create a copy of the repository in your GitHub account.
+
+### Cloning the Repository
+
+1. Open a terminal and clone the `faststore` repository.
+2. Navigate into the project folder.
+
+### Setting Up the Environment
+
+1. Run `yarn` at the root of the repository.
+
+2. Run `yarn build` at the root of the repository.
 
 3. Run `yarn turbo run dev --filter={packageName}` to run a package individually.
 
-> For example, if you want to run the `@faststore/core` package, at the root of the repo run:
+> For example, if you want to run the `@faststore/core` package, at the root of the repository run:
 >
 > `yarn turbo run dev --filter=@faststore/core`
 
-### Developing packages while also testing on a store
+**Note**: If you encounter issues during the setup process, please check if your current version of Yarn, Turbo, and Node.js are matching the versions specified in the package.json file.
+
+## 3. Working on an Issue
+
+### Creating a New Branch
+
+Before starting any work, create a new branch for your changes:
+
+```
+git checkout -b feat/your-branch-name
+```
+
+Choose a branch name that reflects the work being done, such as `fix/query-typo` or `feat/add-new-component`.
+
+### Testing Changes on a Store
 
 To test your changes in a store, you will need to create a pull request (for more guidance, refer to the next section). You can keep your pull request in `draft` while testing.
 
-1. After committing your changes, push them to the remote repo and open a Pull Request.
+1. After committing your changes, push them to the remote repository and open a Pull Request.
 
 2. In the checks section, find `ci/codesandbox` and click on `Details`.
 
```

Comment:
non-blocking: maybe we could change the image below to use faststore/cli instead of the core?

Reply:
Suree! I was about to update it, thanks for the reminder 🌟

Reply:
Updated. [2e196d5](https://github.com/vtex/faststore/pull/2476/commits/2e196d557f71a9f2233370237ffd89232234e792).